File: src/main/java/hub/gateway/shared/AliYunConfig.java
package hub.gateway.shared;

import com.alicloud.openservices.tablestore.AsyncClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

@Component
public class AliYunConfig{
    @Value("${aliyun.accountId}")
    private String accountId;

    @Value("${aliyun.accessKeyId}")
    private String accessKeyId;

    @Value("${aliyun.accessKeySecret}")
    private String accessKeySecret;

    @Value("${aliyun.OTS.endpoint}")
    private String otsEndPoint;

    public AsyncClient createClient(){
        String x = this.getInstanceName();

        AsyncClient client = new AsyncClient(
                this.otsEndPoint,
                this.accessKeyId,
                this.accessKeySecret,
                this.getInstanceName()
        );
        return client;
    }

    private String getInstanceName(){
        Pattern rgxInstName = Pattern.compile("^https?://([^\\.]+)\\.");
        Matcher matcher = rgxInstName.matcher(this.otsEndPoint);
        if(matcher.find())
            return matcher.group(1);

        return null;
    }
}

File: src/main/kotlin/hub/gateway/portal/AliYunConfig.kt
package hub.gateway.portal

import com.alicloud.openservices.tablestore.*
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import java.util.regex.Pattern

@Component
class AliYunConfig {
    @Value("\${aliyun.accountId}")
    private lateinit var accountId:String

    @Value("\${aliyun.accessKeyId}")
    private lateinit var accessKeyId:String

    @Value("\${aliyun.accessKeySecret}")
    private lateinit var accessKeySecret:String

    @Value("\${aliyun.OTS.endpoint}")
    private lateinit var epOTS:String

    companion object {
        // 根据阿里云OTS官方解释,整个应用只需要1个
        private var _otsSyncClient:SyncClientInterface? = null
    }

    fun getOTSSyncClient(): SyncClientInterface {
        if(_otsSyncClient === null){
            var name : String? = null
            // endpoint的第1节就是实例名称
            val rgxName = Pattern.compile("https?://([\\w\\-]+)\\.")
            var matcher = rgxName.matcher(epOTS)
            if(matcher.find())
                name = matcher.group(1)

            _otsSyncClient = SyncClient(epOTS, accessKeyId, accessKeySecret, name)
        }

        return _otsSyncClient!!
    }
}